classes/class-log.php

```diff
@@ -87,9 +73,16 @@ public function log( $connector, $message, $args, $object_id, $context, $action,
 			return false;
 		}
 
+		$ip_address = $this->plugin->get_client_ip_address();
+
+		// Fallback to unsafe IP extracted from the request HTTP headers.
+		if ( empty( $ip_address ) ) {
+			$ip_address = $this->plugin->get_unsafe_client_ip_address();
```

Comment:
Keep the previous behaviour for now. Consider removing this fallback in the future releases (as mentioned in the changelog).

Reply:
I was wondering why you opted to do two separate calls in here, as opposed to having the fallback be part of `get_client_ip_address()` instead.

I thought the latter would have been easier for removing the fallback behavior at a later date. Also, it would allow `get_unsafe_client_ip_address()` to be private instead so we don't create an unwanted public interface for this.

Reply:
The reason is we want to check just the "reliable" IP address in some instances with no fallbacks (when showing the admin notice, for example). This also clearly identifies the places where the use of `get_unsafe_client_ip_address()` should be wrapped into a conditional eventually.